src/main/java/io/appium/java_client/android/AndroidMobileOptions.java

```diff
@@ -0,0 +1,1733 @@
+    }
+
+    /**
+     * Set whether to allow installation of a test package which has android:testOnly="true" in the manifest.
+     *
+     * @param bool when true will be able install a test package marked testOnly.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ALLOW_TEST_PACKAGES
+     */
+    public AndroidMobileOptions setAllowTestPackages(boolean bool) {
+        return amend(AndroidMobileCapabilityType.ALLOW_TEST_PACKAGES, bool);
+    }
+
+    /**
+     * Get whether installation of a test package which has android:testOnly="true" in the manifest is allowed.
+     *
+     * @return true if able to install a test package marked testOnly.
+     * @see AndroidMobileCapabilityType#ALLOW_TEST_PACKAGES
+     */
+    public boolean doesAllowTestPackages() {
+        return (boolean) getCapability(AndroidMobileCapabilityType.ALLOW_TEST_PACKAGES);
+    }
+
+    /**
+     * Set the fully qualified instrumentation class.
+     *
+     * @param coverage is the fully qualified instrumentation class.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ANDROID_COVERAGE
+     */
+    public AndroidMobileOptions setAndroidCoverage(String coverage) {
+        return amend(AndroidMobileCapabilityType.ANDROID_COVERAGE, coverage);
+    }
+
+    /**
+     * Get the fully qualified instrumentation class.
+     *
+     * @return String of the fully qualified instrumentation class.
+     * @see AndroidMobileCapabilityType#ANDROID_COVERAGE
+     */
+    public String getAndroidCoverage() {
+        return (String) getCapability(AndroidMobileCapabilityType.ANDROID_COVERAGE);
+    }
+
+    /**
+     * Set the broadcast action which is used to dump coverage into file system.
+     *
+     * @param coverage is the action to dump coverage into file system.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ANDROID_COVERAGE_END_INTENT
+     */
+    public AndroidMobileOptions setAndroidCoverageEndIntent(String coverage) {
+        return amend(AndroidMobileCapabilityType.ANDROID_COVERAGE_END_INTENT, coverage);
+    }
+
+    /**
+     * Get the broadcast action which is used to dump coverage into file system.
+     *
+     * @return String of the broadcast action which is used to dump coverage into file system.
+     * @see AndroidMobileCapabilityType#ANDROID_COVERAGE_END_INTENT
+     */
+    public String getAndroidCoverageEndIntent() {
+        return (String) getCapability(AndroidMobileCapabilityType.ANDROID_COVERAGE_END_INTENT);
+    }
+
+    /**
+     * Set the timeout to wait for a device to become ready after booting.
+     *
+     * @param duration is the number of seconds to wait for a device to become ready after booting.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ANDROID_DEVICE_READY_TIMEOUT
+     */
+    public AndroidMobileOptions setAndroidDeviceReadyTimeout(Duration duration) {
```

Comment:
It looks like this cap is not used anywhere in appium code: https://github.com/appium/appium/search?q=androidDeviceReadyTimeout&type=Code